presto-orc/src/test/java/com/facebook/presto/orc/metadata/statistics/TestMapColumnStatisticsBuilder.java

```diff
@@ -0,0 +1,154 @@
+        builder.addMapStatistics(key1, columnStatistics1);
+        builder.addMapStatistics(key2, columnStatistics2);
+
+        MapColumnStatistics columnStatistics = (MapColumnStatistics) builder.buildColumnStatistics();
+        assertEquals(columnStatistics.getNumberOfValues(), 8);
+
+        MapStatistics mapStatistics = columnStatistics.getMapStatistics();
+        List<MapStatisticsEntry> entries = mapStatistics.getEntries();
+        assertEquals(entries.size(), 2);
+        assertEquals(entries.get(0).getKey(), key1);
+        assertEquals(entries.get(0).getColumnStatistics(), columnStatistics1);
+        assertEquals(entries.get(1).getKey(), key2);
+        assertEquals(entries.get(1).getColumnStatistics(), columnStatistics2);
+    }
+
+    @Test(dataProvider = "keySupplier")
+    public void testMergeMapStatistics(KeyInfo[] keys)
+    {
+        // merge two stats with keys: [k0,k1] and [k1,k2]
+        // column statistics for k1 should be merged together
+        MapColumnStatisticsBuilder builder1 = new MapColumnStatisticsBuilder();
+        builder1.addMapStatistics(keys[0], new ColumnStatistics(3L, null));
+        builder1.addMapStatistics(keys[1], new ColumnStatistics(5L, null));
+        ColumnStatistics columnStatistics1 = builder1.buildColumnStatistics();
+
+        MapColumnStatisticsBuilder builder2 = new MapColumnStatisticsBuilder();
+        builder2.addMapStatistics(keys[1], new ColumnStatistics(7L, null));
+        builder2.addMapStatistics(keys[2], new ColumnStatistics(9L, null));
+        ColumnStatistics columnStatistics2 = builder2.buildColumnStatistics();
+
+        MapStatistics mergedMapStatistics = MapColumnStatisticsBuilder.mergeMapStatistics(ImmutableList.of(columnStatistics1, columnStatistics2)).get();
+        assertMergedMapStatistics(keys, mergedMapStatistics);
+    }
+
+    @Test(dataProvider = "keySupplier")
+    public void testMergeMapStatisticsMissingStats(KeyInfo[] keys)
+    {
+        // valid map stat
+        MapColumnStatisticsBuilder builder1 = new MapColumnStatisticsBuilder();
+        builder1.addMapStatistics(keys[0], new ColumnStatistics(3L, null));
+        ColumnStatistics columnStatistics1 = builder1.buildColumnStatistics();
+
+        // invalid map stat
+        ColumnStatistics columnStatistics2 = new ColumnStatistics(7L, null);
+
+        Optional<MapStatistics> mergedMapStats = MapColumnStatisticsBuilder.mergeMapStatistics(ImmutableList.of(columnStatistics1, columnStatistics2));
+        assertFalse(mergedMapStats.isPresent());
+    }
+
+    @Test(dataProvider = "keySupplier")
+    public void testMergeColumnStatistics(KeyInfo[] keys)
+    {
+        // merge two stats with keys: [k0,k1] and [k1,k2]
+        // column statistics for k1 should be merged together
+        MapColumnStatisticsBuilder builder1 = new MapColumnStatisticsBuilder();
+        builder1.addMapStatistics(keys[0], new ColumnStatistics(3L, null));
```

Comment:
Can we have some IntegerStatistics with min/max and see how they are merged ? Or is IntegerStats not supported here.

Reply:
Great idea. Updated the test to use more specific column stats.